### PR TITLE
Governance vote

### PR DESCRIPTION
**Pull request by NovacAndrei**

### CODE REVIEW

File: src/core/blockchain/types/transaction.ts [hunk 1]
```diff
@@ -137,5 +137,6 @@ export enum TransactionMessageText {
     CONTRACT_TX_NORMAL_NOT_ALLOWED = 'CONTRACT_TX_NORMAL_NOT_ALLOWED',
     WAITING_TX_CONFIRM = 'WAITING_TX_CONFIRM',
     WAITING_TX_CONFIRM_CANCEL = 'WAITING_TX_CONFIRM_CANCEL',
-    INSUFFICIENT_FUNDS_SOURCE_ACCOUNT = 'INSUFFICIENT_FUNDS_SOURCE_ACCOUNT'
+    INSUFFICIENT_FUNDS_SOURCE_ACCOUNT = 'INSUFFICIENT_FUNDS_SOURCE_ACCOUNT',
+    SENDING_VOTE = 'SENDING_VOTE'
```
Comment (georgeburduhos) [inline src/core/blockchain/types/transaction.ts:141]: this might be confusing in celo where is vote. Maybe put 

`SENDING_GOVERNNANCE_VOTE`

**Review by georgeburduhos — APPROVED** [review]

File: src/core/blockchain/types/transaction.ts (hunk 1 above)
Comment (krisboit) [reply src/core/blockchain/types/transaction.ts:141]: yeah, I agree, maybe `GOVERNANCE_VOTE` would be more suitable.

File: src/redux/ui/screens/data/handle-cta.ts [hunk 2]
```diff
@@ -835,6 +849,125 @@ const handleCtaAction = async (
...
+
+                    const wallet: {
+                        signMessage: (
+                            blockchain: Blockchain,
+                            accountIndex: number,
+                            accountType: AccountType,
+                            message: string
+                        ) => Promise<any>;
+                    } =
+                        appWallet.type === WalletType.HW
```
Comment (krisboit) [inline src/redux/ui/screens/data/handle-cta.ts:889]: we need to test if message signing is working on ledger, this was not testet on SDK implementation

File: src/redux/ui/screens/data/handle-cta.ts [hunk 3]
```diff
@@ -835,6 +849,125 @@ const handleCtaAction = async (
...
+                    const appWallet = getWalletByPubKey(state, selectedWallet.walletPublicKey);
+
+                    if (!appWallet || !account) {
+                        break;
+                    }
+
+                    let password = '';
+
+                    if (appWallet.type === WalletType.HD) {
+                        password = await PasswordModal.getPassword(
```
Comment (krisboit) [inline src/redux/ui/screens/data/handle-cta.ts:870]: all awaits should be in a try...catch

**Review by krisboit — CHANGES_REQUESTED** [review]

File: src/components/smart-screen/smart-screen.tsx [hunk 4]
```diff
@@ -158,6 +158,7 @@ class SmartScreenComp extends React.Component<IReduxProps & IExternalProps, ISta
                     setScreenInputData: this.props.setScreenInputData
                 }}
                 blockchain={this.props.account.blockchain}
+                scrollPubSub={undefined}
```
Comment (krisboit) [inline src/components/smart-screen/smart-screen.tsx:161]: change this to pubSub, we could send here different kind of events not only scroll events

**Review by krisboit — CHANGES_REQUESTED** [review]

**Review by krisboit — APPROVED** [review]

**Pull request merged**